app/src/cli/run.rs

```diff
@@ -16,10 +16,14 @@ pub fn exec(cmd: Args) -> miette::Result<()> {
     let mut db = Database::default();
     let file = File::read(&cmd.filepath).map_err(IOError::from).map_err(miette::Report::from)?;
     let view = db.add(file).query();
-    let nf = view.run().map_err(|err| view.pretty_error(err))?;
     println!("{} typechecked successfully!", cmd.filepath.display());
-    if let Some(nf) = nf {
-        print_nf(&nf)
+
+    let nf = view.run().map_err(|err| view.pretty_error(err))?;
+    match nf {
+        Some(nf) => print_nf(&nf),
+        None => {
+            println!("No \"main\" expression was found in the program.")
```

Comment:
`eprintln!` to print to stderr or better `return Err(...)` to print to stderr and exit with status code 1.

Reply:
As discussed, we should split `pol run` into `pol check` and `pol run`. I can do that in a separate PR.

Reply:
I have implemented both `check` and `run` in this PR, and use a miette error to report if a main expression wasn't found.